### PR TITLE
#1959. Test valid combinations of modifiers. Part 3

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Class-modifiers/syntax_base_mixin_A06_t01.dart [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      ImplementsM _ => "ImplementsM",
+      WithM _ => "WithM"
+    };
+
+String test2(BaseMixin c) => switch (c) {
+//                           ^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+      ImplementsBaseMixin _ => "ImplementsBaseMixin",
+      WithBaseMixin _ => "WithBaseMixin"
```
Comment (eernstg) [inline LanguageFeatures/Class-modifiers/syntax_base_mixin_A06_t01.dart:32]: I wonder why this one wasn't flagged as unreachable...

File: LanguageFeatures/Class-modifiers/syntax_mixin_A02_t01.dart [hunk 2]
```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Mixin can be implemented and mixed in but not constructed or
+/// extended and is not exhaustive
+///
+/// @description Checks that it is a compile-time error to construct a `mixin`
```
Comment (eernstg) [inline LanguageFeatures/Class-modifiers/syntax_mixin_A02_t01.dart:8]: Perhaps?:
```suggestion
/// @description Checks that it is a compile-time error to extend a `mixin`
```

File: LanguageFeatures/Class-modifiers/syntax_mixin_A02_t02.dart [hunk 3]
```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Mixin can be implemented and mixed in but not constructed or
+/// extended and is not exhaustive
+///
+/// @description Checks that it is a compile-time error to construct a `mixin`
```
Comment (eernstg) [inline LanguageFeatures/Class-modifiers/syntax_mixin_A02_t02.dart:8]:
```suggestion
/// @description Checks that it is a compile-time error to extend a `mixin`
```

File: LanguageFeatures/Class-modifiers/syntax_mixin_A06_t01.dart [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+mixin _M {}
+class ImplementsM implements _M {}
+class WithM with _M {}
+
+String test1(_M c) => switch (c) {
+//                    ^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+      ImplementsM _ => "ImplementsM",
+      WithM _ => "WithM"
```
Comment (eernstg) [inline LanguageFeatures/Class-modifiers/syntax_mixin_A06_t01.dart:24]: Again just wondering why we don't get the UNREACHABLE message here.

File: LanguageFeatures/Class-modifiers/syntax_mixin_class_A03_t02.dart [hunk 5]
```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Mixin class can be constructed, extended, implemented and mixed
+/// in but is not exhaustive
+///
+/// @description Checks that a `mixin class` can be mixed in it the same
```
Comment (eernstg) [inline LanguageFeatures/Class-modifiers/syntax_mixin_class_A03_t02.dart:8]:
```suggestion
/// @description Checks that a `mixin class` can be mixed in in the same
```

File: LanguageFeatures/Class-modifiers/syntax_base_mixin_A06_t01.dart (hunk 1 above)
Comment (eernstg) [reply LanguageFeatures/Class-modifiers/syntax_base_mixin_A06_t01.dart:32]: Ah, of course: The UNREACHABLE one was `sealed`, that's the reason why it was known that there couldn't be any instances. With this one we could actually have an instance of `WithM`.

File: LanguageFeatures/Class-modifiers/syntax_mixin_A06_t01.dart (hunk 4 above)
Comment (eernstg) [reply LanguageFeatures/Class-modifiers/syntax_mixin_A06_t01.dart:24]: (Closing this thread, the point was that the UNREACHABLE cases were `sealed` classes.)

**Review by eernstg — COMMENTED** [review]
Looks good! Just a couple of typos in 'description'.

**Comment by sgrekhov** [comment]
Typos fixed. Please take another look

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**